samza-core/src/main/java/org/apache/samza/config/BlobStoreConfig.java

```diff
@@ -24,18 +24,59 @@
  */
 public class BlobStoreConfig extends MapConfig {
 
-  public static final String BLOB_STORE_MANAGER_FACTORY = "blob.store.manager.factory";
-  public static final String BLOB_STORE_ADMIN_FACTORY = "blob.store.admin.factory";
+  private static final String PREFIX = "blob.store.";
+  public static final String BLOB_STORE_MANAGER_FACTORY = PREFIX + "manager.factory";
+  public static final String BLOB_STORE_ADMIN_FACTORY = PREFIX + "admin.factory";
+  // Configs related to retry policy of blob stores
+  private static final String RETRY_POLICY_PREFIX = PREFIX + "retry.policy.";
+  public static final String RETRY_POLICY_MAX_RETRIES = RETRY_POLICY_PREFIX + "max.retries";
+  // -1 for RetryPolicy means unlimited retries. Retry is limited by max retry duration, rather than count of retries.
+  public static final int DEFAULT_RETRY_POLICY_MAX_RETRIES = -1;
+  public static final String RETRY_POLICY_MAX_RETRY_DURATION_MILLIS = RETRY_POLICY_PREFIX + "max.retry.duration.millis";
+  public static final long DEFAULT_RETRY_POLICY_MAX_RETRY_DURATION_MILLIS = 10 * 60 * 1000; // 10 mins
+  public static final String RETRY_POLICY_BACKOFF_DELAY_MILLIS = RETRY_POLICY_PREFIX + "backoff.delay.millis";
+  public static final long DEFAULT_RETRY_POLICY_BACKOFF_DELAY_MILLIS = 100;
+  public static final String RETRY_POLICY_BACKOFF_MAX_DELAY_MILLIS = RETRY_POLICY_PREFIX + "backoff.max.delay.millis";
+  public static final long DEFAULT_RETRY_POLICY_BACKOFF_MAX_DELAY_MILLIS = 312500;
+  public static final String RETRY_POLICY_BACKOFF_DELAY_FACTOR = RETRY_POLICY_PREFIX + "backoff.delay.factor";
+  public static final int DEFAULT_RETRY_POLICY_BACKOFF_DELAY_FACTOR = 5;
+  public static final String RETRY_POLICY_JITTER_FACTOR_MILLIS =  RETRY_POLICY_PREFIX + "jitter.factor.millis";
+  // random retry delay between -100 to 100 millisecond
+  public static final long DEFAULT_RETRY_POLICY_JITTER_FACTOR_MILLIS = 100;
```

Comment:
This is factor, not amount, right? Is this defaulting to a factor of 100?

Reply:
Also fix comment above.

Reply:
This is actually amount -> e.g. default of 100 ms. Should be names just jitter instead of jitter factor. Updating it.

Reply:
Prefer using factor instead of fixed (inlike the current actual impl), since it scales better with longer retry duration. E.g. a facor of 0.1 - 0.2 works even at a retry interval of ~5 min.